lisp/Makefile

```diff
@@ -51,6 +51,11 @@ uninstall:
 distclean clean:
 	rm -f $(ELC) ess-autoloads.el julia-mode.el .dependencies
 
+.PHONY: test
+test:
+	! ( $(COMPILE) $(shell git ls-files *.el) 2>&1 | awk '{if (/^ /) { gsub(/^ +/, " ", $$0); printf "%s", $$0 } else { printf "\n%s", $$0 }}' | egrep -a "not known|Error|free variable|error for|Use of gv-ref" )
```

Comment:
Why this new target?

Also, this line is way too long. Please cut lines around 80ish characters.

Reply:
I almost have to insist on the lisp compiling cleanly.  It found the error in #748

Reply:
This isn't the right way to test for byte compilation warnings. You'd want to set `byte-compile-error-on-warn` to `t` while compiling.

We're already doing that for Emacs 26+ on travis. I was the one responsible for fixing several hundred byte compiler warnings and I don't use Emacs 25 but we should probably enable it there too. I'll look into grabbing an old Emacs and seeing what needs fixing.

Reply:
ah, okay.  i knew there had to be a better way.